webrtc.html

```diff
@@ -1837,61 +1808,32 @@ <h4>Set the RTCSessionDescription</h4>
                             <p>Set <var>transceiver</var>.<a>[[\Receiver]]</a>.<a>[[\ReceiveCodecs]]</a>
                             to <var>transceiver</var>.<a>[[\Receiver]]</a>.<a>[[\LastStableStateReceiveCodecs]]</a>.</p>
                           </li>
-                          <li data-tests="RTCPeerConnection-setDescription-transceiver.html">
-                            <p>If the <var>transceiver</var> was created by applying the
-                            <code><a>RTCSessionDescription</a></code> that is
-                            being rolled back, and a track has never been attached to
-                            it via <code>addTrack</code>, run the following steps:</p>
-                            <ol>
-                              <li>
-                                <p>If the <var>transceiver</var>'s <a>[[\FiredDirection]]</a>
-                                is either <code>"sendrecv"</code> or <code>"recvonly"</code>,
-                                <a>process the removal of a remote track</a> with
-                                <var>transceiver</var> and <var>muteTracks</var> and set
-                                <a>[[\FiredDirection]]</a> to <code>"inactive"</code>.</p>
-                              </li>
-                              <li>
-                                <p><a data-lt="stop the RTCRtpTransceiver">Stop the
-                                RTCRtpTransceiver</a> <var>transceiver</var>.</p>
-                              </li>
-                              <li>
-                                <p>Remove <var>transceiver</var> from <var>connection</var>'s
-                                <a href="#transceivers-set">set of transceivers</a>.</p>
-                              </li>
-                            </ol>
-                          </li>
                           <li>
-                            <p>Otherwise, (if the <var>transceiver</var> was not just removed)
-                            run the following steps:</p>
+                            <p>If <var>remote</var> is <code>true</code>, run the following sub steps:</p>
```

Comment:
Do we set "remote" correctly for "rollback"? setLocalDescription() with rollback and setRemoteDescription() with rollback  should behave exactly the same. I don't think we set "remote" to reflect which description is being rolled back?

In setRemoteDescription's implicit rollback we say "Let p be the result of setting the local RTCSessionDescription indicated by {type: "rollback"}." It makes sense that "remote" is false in this case, because only local descriptions will implicitly get rolled back.

In setLocalDescription, unless I'm reading this mistaken, if we explicitly perform "rollback" then we end up at "return the result of setting the local RTCSessionDescription indicated by {type, sdp}." So in this case, when we explicitly rollback, we set "remote" to false regardless if we're rolling back a local or remote offer.

Is this a bug? Should we do one of the following:
1. In all rollback cases, make sure "remote" reflects the offer being rolled back, not whether SLD or SRD is called.
2. Instead of saying "if remote is true" here, saying "if signalingState is have-remote-offer"?

Reply:
Good catch! I picked 2 since it looks like we don't read _remote_ anywhere but here on rollback.